### PR TITLE
Alerting: Use the forked Alertmanager for remote secondary mode

### PR DESCRIPTION
**Pull request by santihernandezc**
This PR wires everything up to use the forked Alertmanager. It reads the configuration options and feature toggles to decide how to start Grafana.

If the remote alertmanager is enabled, one of the three modes for the remote Alertmanager should be enabled via feature toggles too. The only mode we support right now is _remote secondary_, so enabling the feature toggles for _remote primary_ or _remote only_ results in Grafana falling back to _remote secondary_. Not enabling any of these feature toggles results in Grafana falling back to using only the internal Alertmanager.

I've added a configurable `syncInterval`, which is the time we wait between consecutive syncs with the remote Alertmanager. When this interval elapses, we send our configuration and state from the Grafana Alertmanager to the remote one.

### CODE REVIEW

File: pkg/services/ngalert/ngalert.go [hunk 1]
```diff
@@ -173,16 +173,45 @@ func (ng *AlertNG) init() error {
...
+					URL:               ng.Cfg.UnifiedAlerting.RemoteAlertmanager.URL,
+					TenantID:          ng.Cfg.UnifiedAlerting.RemoteAlertmanager.TenantID,
+					BasicAuthPassword: ng.Cfg.UnifiedAlerting.RemoteAlertmanager.Password,
+				}
+				// We won't be handling files on disk, we can pass an empty string as workingDirPath.
+				stateStore := notifier.NewFileStore(orgID, ng.KVStore, "")
+				r, err := remote.NewAlertmanager(externalAMCfg, orgID, stateStore)
+				if err != nil {
+					return nil, err
+				}
```
Comment (gotjosh) [inline pkg/services/ngalert/ngalert.go:201]: I don't think we care about errors on secondary in this mode.

File: pkg/setting/setting_unified_alerting.go [hunk 2]
```diff
@@ -352,6 +353,11 @@ func (cfg *Cfg) ReadUnifiedAlertingSettings(iniFile *ini.File) error {
 		TenantID: remoteAlertmanager.Key("tenant").MustString(""),
 		Password: remoteAlertmanager.Key("password").MustString(""),
 	}
+	uaCfgRemoteAM.SyncInterval, err = gtime.ParseDuration(valueAsString(remoteAlertmanager, "sync_interval", (schedulerDefaultAdminConfigPollInterval).String()))
```
Comment (gotjosh) [inline pkg/setting/setting_unified_alerting.go:356]: This should be done directly on the configuration struct.

File: conf/defaults.ini [hunk 3]
```diff
@@ -1261,19 +1261,22 @@ clean_upgrade = false
 
 # Enable the use of the configured remote Alertmanager and disable the internal one.
 # The default value is `false`.
```
Comment (gotjosh) [inline conf/defaults.ini:1263]: You're probably testing but I don't think there should be changes on this file - let's aim for not documenting these fields in the `ini` files for now - opt for documenting them in the settings struct.

**Review by gotjosh — COMMENTED** [review]
Looking good - I know this is a draft but let you a few comments.

File: pkg/setting/setting_unified_alerting.go (hunk 2 above)
Comment (santihernandezc) [reply pkg/setting/setting_unified_alerting.go:356]: All intervals taken from the .ini file are parsed [here](https://github.com/grafana/grafana/blob/fdaf6e3f2ecfe379350884f9da4e2841980da037/pkg/setting/setting_unified_alerting.go#L232-L251)

Comment (santihernandezc) [reply pkg/setting/setting_unified_alerting.go:356]: Unless we have a really good reason I don't think we should take a different approach.

Comment (gotjosh) [reply pkg/setting/setting_unified_alerting.go:356]: Oh sorry, I missed this was done on the settings package 🤦 - you're good, carry on.

File: pkg/services/ngalert/remote/alertmanager_test.go [hunk 4]
```diff
@@ -159,7 +159,7 @@ func TestIntegrationRemoteAlertmanagerApplyConfigOnlyUploadsOnce(t *testing.T) {
 	silences := []byte("test-silences")
 	nflog := []byte("test-notifications")
 	store := fakes.NewFakeKVStore(t)
-	fstore := notifier.NewFileStore(1, store, "")
+	fstore := notifier.NewFileStore(cfg.OrgID, store, "")
```
Comment (santihernandezc) [inline pkg/services/ngalert/remote/alertmanager_test.go:162]: Unrelated, I'm using the value in the config struct instead of a hard-coded `1`

**Review by gotjosh — APPROVED** [review]
Beautiful!

**Pull request merged**